django/core/management/commands/shell.py

```diff
@@ -168,20 +202,20 @@ def get_and_report_namespace(self, **options):
             import_string = isort.code(import_string)
 
         if import_string:
-            msg = f"{msg}, including:\n\n{import_string}"
+            msg = f"{msg}:\n\n{import_string}"
         else:
             msg = f"{msg}."
 
         self.stdout.write(msg, self.style.SUCCESS, ending="\n\n")
 
-        return namespace
+        if errors:
+            msg = "\n".join(f"  {e}" for e in import_errors)
+            self.stdout.write(
+                f"{errors} objects could not be automatically imported:\n\n{msg}",
```

Comment:
(nit) I think we will have the same issue of "1 objects" here

Reply:
Great catch, fixed with tests.